### PR TITLE
#692 - Extracted FileProxy and supported auth

### LINKED ISSUE #692: Authentication for files proxy remote
Files proxy should be configurable to specify username and password for target repository

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #692 
Extracted `FileProxy` and supported auth

### CODE REVIEW

File: src/main/java/com/artipie/file/FileProxy.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Map;
+import org.reactivestreams.Publisher;
+
+/**
+ * File proxy slice created from config.
+ *
+ * @since 0.12
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/file/FileProxy.java:42]: @olegmoz is this suppression necessary here? I'd suggest not to add it if not, as generally this [check](https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.html) is connected to the complexity of the class and when we have to add this suppression it's the first signal to required decomposition.

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz just one comment, please, take a look

File: src/main/java/com/artipie/file/FileProxy.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artipie/file/FileProxy.java]: @olenagerasimova that suppression was here by mistake, I totally agree that this warning is meaningful in production code

**Review by olenagerasimova — APPROVED** [review]
@olegmoz thanks, good to merge

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#707` is not assigned, can't get performer

**Pull request merged**